lib/Db/QueueMapper.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace OCA\Recognize\Db;
+
+use OCA\Recognize\Classifiers\Audio\MusicnnClassifier;
+use OCA\Recognize\Classifiers\Images\ClusteringFaceClassifier;
+use OCA\Recognize\Classifiers\Images\ImagenetClassifier;
+use OCA\Recognize\Classifiers\Images\LandmarksClassifier;
+use OCA\Recognize\Classifiers\Video\MovinetClassifier;
+use OCP\AppFramework\Db\QBMapper;
+use OCP\DB\QueryBuilder\IQueryBuilder;
+use OCP\IDBConnection;
+
+class QueueMapper extends QBMapper {
+	public const MODELS = [
+		ImagenetClassifier::MODEL_NAME,
+		ClusteringFaceClassifier::MODEL_NAME,
+		LandmarksClassifier::MODEL_NAME,
+		MovinetClassifier::MODEL_NAME,
+		MusicnnClassifier::MODEL_NAME
+	];
+
+	/**
+	 * @var IDBConnection $db
+	 */
+	protected $db;
+
+	public function __construct(IDBConnection $db) {
+		parent::__construct($db, '', QueueFile::class);
```

Comment:
Since the table is not passed maybe it would be good to implement QBMappers getTableName to still return that in case one of the parent methods like insert/update/delete  are used

Reply:
This is tough, because the table name is constructed on demand. Not sure how an implementation of getTableName would look like. This class is a bit of a hack admittedly. I just didn't want to have a queue class for every model type...

Reply:
or do you mean getTableName should throw?

Reply:
Yes, throwing would also be ok